dev/src/bulk-writer.ts

```diff
@@ -714,27 +739,39 @@ export class BulkWriter {
     const tag = requestTag();
     const pendingBatch = this._bulkCommitBatch;
 
-    this._bulkCommitBatch = new BulkCommitBatch(this.firestore);
-
-    // Send the batch if it is under the rate limit, or schedule another
-    // attempt after the appropriate timeout.
-    const underRateLimit = this._rateLimiter.tryMakeRequest(
+    // Use the write with the highest failure count when determining backoff.
+    const highestFailureCount = pendingBatch.pendingOps.reduce((prev, cur) =>
+      prev.failedAttempts > cur.failedAttempts ? prev : cur
+    ).failedAttempts;
+    const resourceExhausted =
```

Comment:
Not sure if this is worth adding a getter for `pendingOps`, or if I should just add instance methods on `BulkCommitBatch` for these two usages.